lib/h265dec.cpp

```diff
@@ -561,17 +1020,24 @@ gst_h265_dec_update_picture_parameters (GstH265Decoder * decoder,
     }
     case GST_H265_NAL_VPS:{
       GstH265VPS *vps = static_cast < GstH265VPS * >(nalu);
-      // if (pps_cmp (&self->last_pps, pps))
-      //   return;
-      // self->last_pps = *pps;
-      // fill_vps (pps, &self->vkp);
+      if (vps_cmp (&self->last_vps, vps))
+         return;
+      self->last_vps = *vps;
+      /* FIXME: The VPS arrives first but the vk video decoder expects the SPS to be provided first. 
```

Comment:
Does this mean that currently no VPS nal is passed to decoder?

Reply:
it's ok if it's not, just to be aware of it.

Reply:
yes the VPS is discarded by the NV parser, so I decided to remove it. That's why I put a FIXME

Reply:
If the VPS is processed then vk_video_samples is complaining that the SPS has not been received and assert